### PR TITLE
Support S32 audio sample formats

### PR DESCRIPTION
**Pull request by Splendide-Imaginarius**
Fixes FLAC playback; not sure what other formats it might fix.

### CODE REVIEW

**Comment by Splendide-Imaginarius** [comment]
This doesn't work yet. I'll come back to this later.

File: src/audio/aldatasource.h [hunk 1]
```diff
@@ -56,7 +56,8 @@ struct ALDataSource
 ALDataSource *createSDLSource(SDL_RWops &ops,
                               const char *extension,
 			                  uint32_t maxBufSize,
-			                  bool looped);
+			                  bool looped,
+			                  int fallbackMode);
```
Comment (Ancurio) [inline src/audio/aldatasource.h:60]: Any reason this is an int and not a bool?

**Comment by Splendide-Imaginarius** [comment]
This is being worked on at upstream OpenAL Soft, so this PR might be unnecessary soon. Marking as draft until it's more clear what OpenAL Soft will do.

**Comment by Splendide-Imaginarius** [comment]
OpenAL Soft isn't moving yet on a fix, and it looks to me like this PR is necessary anyway for foreign-endian edge cases, so I'm undrafting it.

**Comment by Splendide-Imaginarius** [comment]
@Eblo would you be able to test whether this PR fixes FLAC playback?

**Comment by Eblo** [comment]
I can confirm this fixes/enables FLAC playback.

**Comment by Splendide-Imaginarius** [comment]
Thanks, I'll merge this soon.

**Comment by Ancurio** [comment]
Sorry for the late comment, I didn't notice Github held it in "Pending" state and didn't actually publish it

File: src/audio/aldatasource.h (hunk 1 above)
Comment (Splendide-Imaginarius) [reply src/audio/aldatasource.h:60]: Yes, the intent is that this logic should be extensible in case a 2nd fallback attempt is needed; the 2nd fallback would be mode 2.
